### ISSUE TITLE
Add option to include symbols for runtime packs when running ILLink

### ISSUE THREAD

**Issue by vitek-karas** [dotnet/sdk#12962]
Currently when the SDK runs ILLink it will only produce updated pdbs for assemblies which it has pdbs for. Which basically excludes all framework assemblies (from runtime packs). But ILLink will modify the framework assemblies. So after trimming there's no way to get the new pdbs for the trimmed framework anymore.

There should be an option (MSBuild property) which will tell SDK to gather symbols for everything it can before running ILLink so that trimming can also produce matching PDBs and allow full debugging experience across all assemblies.

**Comment by sbomer** [comment]
The way I would imagine this working is that we would ship [`Microsoft.SymbolStore`](https://github.com/dotnet/symstore) in the SDK with a task that can fetch symbols matching the framework version we are publishing (and maybe also third-party nugets?). It could be useful independently of the linker, so maybe the option could be called `PublishWithSymbols` or similar.

@mikem8361 @wli3 does this seem like a reasonable approach?

**Comment by mikem8361** [comment]
Yes, the Microsoft.SymbolStore code could be used to create a download symbol msbuild task. What I'm confused about is what PDB is being downloaded: the original PDB for the untrimmed assembly or the PDB for the trimmed ones? What get published to the symbol server?

**Comment by vitek-karas** [comment]
We would download the untrimmed symbols - the exact same one which are already published to symbol server today. The trimmed ones are produced locally when trimming the app. Trimming the app trims framework assemblies - for that one app alone - so it will produce a "trimmed" version of the symbols locally. In order to do that, it needs the original untrimmed symbols (just like it consumes the untrimmed assemblies on input).

**Comment by mikem8361** [comment]
I understand. It wouldn't take much for someone to write a msbuild task using Microsoft.SymbolStore to do this download.

**Comment by wli3** [comment]
@KathleenDollard @dsplaisted Microsoft.SymbolStore is only 71KB. And the scenario is very useful. I think we should go ahead. What do you think?

**Comment by dsplaisted** [comment]
Seems reasonable to me.

**Comment by mikem8361** [comment]
FYI Microsoft.SymbolStore does have a dependency on Microsoft.FileFormats (99K)

**Comment by vitek-karas** [comment]
We should do this in such a way that it also happens for NativeAOT, since it has the same problem.

**Comment by agocke** [comment]
@marcpopMSFT retriaging back to area-tools. As mentioned in the linked issue, this would be useful for ILLink/NativeAOT (they would consume the PDBs if present) but it doesn't have anything to do with the tools themselves and can't be implemented by them directly (since they cannot rely on network access).

**Comment by marcpopMSFT** [comment]
Triage: @agocke this is specifically an ask to have restore download the symbol nuget packages? Is the nuget issue tracking this sufficient or is there work in the SDK as well?

**Comment by marcpopMSFT** [comment]
Ping @agocke since it's been a week.

**Comment by agocke** [comment]
It could be NuGet, as long as whatever's in NuGet makes it into the SDK. The core feature we need is some sort of Task/Target built into the SDK that can satisfy these requests.

**Comment by marcpopMSFT** [comment]
@agocke where are the symbols being published today? I assume symbol server but not nuget.org. 

Checking with @nkolev92 to comment on how this would work where restore would try to look up the symbol server info and download the packages (or would that be post download in the SDK).

**Comment by nkolev92** [comment]
There's an issue related to this topic that was filled a while back: https://github.com/nuget/home/issues/9667. 

Symbol packages not really downloadable today, pdbs are, so there's really no place symbol packages are store in a way easy to download today. They might as well not exist after they've been uploaded anywhere. 

The client itself has never really talk to symbol servers beyond a single push endpoint, so it's only been a one directional communication so far. 

We probably need to have some more conversations to make progress on the overall story.

**Comment by baronfel** [comment]
How does the VS Debugger/VSCodedbg get symbols for the runtime libraries today? Is that a public API/Client?
